crates/uv-git/src/git.rs

```diff
@@ -133,24 +124,69 @@ pub(crate) struct GitRemote {
 /// A local clone of a remote repository's database. Multiple [`GitCheckout`]s
 /// can be cloned from a single [`GitDatabase`].
 pub(crate) struct GitDatabase {
-    /// The remote repository where this database is fetched from.
-    remote: GitRemote,
-    /// Path to the root of the underlying Git repository on the local filesystem.
-    path: PathBuf,
     /// Underlying Git repository instance for this database.
-    repo: git2::Repository,
+    repo: GitRepository,
 }
 
-/// A local checkout of a particular revision from a [`GitDatabase`].
-pub(crate) struct GitCheckout<'a> {
-    /// The git database where this checkout is cloned from.
-    database: &'a GitDatabase,
-    /// Path to the root of the underlying Git repository on the local filesystem.
-    path: PathBuf,
+/// A local checkout of a particular revision from a [`GitRepository`].
+pub(crate) struct GitCheckout {
     /// The git revision this checkout is for.
-    revision: git2::Oid,
+    revision: GitOid,
     /// Underlying Git repository instance for this checkout.
-    repo: git2::Repository,
+    repo: GitRepository,
+}
+
+/// A local Git repository.
+pub(crate) struct GitRepository {
+    /// Path to the underlying Git repository on the local filesystem.
+    path: PathBuf,
+}
+
+impl GitRepository {
+    /// Opens an existing Git repository at `path`.
+    pub(crate) fn open(path: &Path) -> Result<GitRepository> {
+        // Make sure there is a Git repository at the specified path.
+        ProcessBuilder::new("git")
+            .arg("rev-parse")
+            .cwd(path)
+            .exec_with_streaming(&mut silent, &mut silent, true)?;
+
+        Ok(GitRepository {
+            path: path.to_path_buf(),
+        })
+    }
+
+    /// Initializes a Git repository at `path`.
+    fn init(path: &Path) -> Result<GitRepository> {
+        // TODO: see if this still necessary
```

Comment:
Nit: can you write out your TODOs as `TODO(ibraheem): ...` or `TODO(ibraheemdev): ...` or whatever you prefer? I like having an author annotation on TODOs -- it doesn't necessarily mean that you're responsible for _solving_ it, but it makes it immediately clear who has context on it.